plugin/metricstore/prometheus/metricstore/reader.go

```diff
@@ -78,10 +95,27 @@ func NewMetricsReader(cfg config.Configuration, logger *zap.Logger, tracer trace
 		return nil, fmt.Errorf("failed to initialize prometheus client: %w", err)
 	}
 
+	customClient := promClient{
+		Client: client,
+		params: cfg.AdditionalParameters,
+	}
+
+	return customClient, nil
+}
+
+// NewMetricsReader returns a new MetricsReader.
+func NewMetricsReader(cfg config.Configuration, logger *zap.Logger, tracer trace.TracerProvider) (*MetricsReader, error) {
+	logger.Info("Creating metrics reader", zap.Any("configuration", cfg))
```

Comment:
not a good idea to log configuration as it may contain credentials

Reply:
This code was already present before I refactored it. Removed it as suggested.